sr_data_visualization/src/sr_data_visualization/data_plot.py

```diff
@@ -115,23 +143,93 @@ def show_trace(self, trace_name):
             else:
                 trace.plot.detach()
 
+    def clear_data(self):
+        for trace in self.traces:
+            trace.data = np.zeros(self.x_data.shape)
+            trace.latest_value = 0.0
+            trace.plot.setData(self.x_data, trace.data)
+        self.replot()
+
 
 class JointStatesDataPlot(GenericDataPlot):
     def __init__(self, joint_name, topic_name, topic_type):
         super().__init__(joint_name, topic_name, topic_type, start_plotting=True)
 
+        self.side = self.joint_name[0:2]
+
     def create_traces(self):
+        self.buffers_left = [DataBuffer(self.x_data.shape),
+                             DataBuffer(self.x_data.shape),
+                             DataBuffer(self.x_data.shape)]
+
+        self.buffers_right = [DataBuffer(self.x_data.shape),
+                              DataBuffer(self.x_data.shape),
+                              DataBuffer(self.x_data.shape)]
```

Comment:
Do we need to have buffers for left and right if, at any given point of time, we're only displaying one, and before replotting we clear the buffer, so no data remains inside?

Reply:
Right now the only moment in which the data isn't cleared is when the tab of Joint States is shown and you change the side, that way you can see instantly the two plots (left/right) completed.

Reply:
so the behavior should be consistent, if I switch sides on Control Loop, MotorStats, and so on, it will start collecting/plotting the data from a clear buffer, for JointStates it keeps the history.

Reply:
exactly, if you think it should be consistent I can just remove the buffers from Joint States. If I do a similar behaviour for Control Loop and Palm Extras I think there would be too many subscriptions (left and right).